middleware/timeout_test.go

```diff
@@ -0,0 +1,90 @@
+// +build go1.13
+
+package middleware
+
+import (
+	"errors"
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestTimeout(t *testing.T) {
```

Comment:
Please add a test that checks that the request is properly cloned.
Maybe you can use this Context methods that relies on the inner http.Request
* QueryString()
* FormValue(name)
* Cookie(name)

Reply:
I will add, but do we really want to be testing the functionality of Go core itself? https://github.com/golang/go/blob/release-branch.go1.13/src/net/http/request.go#L370-L393

Reply:
The intention here is probably to ensure we have the correct request (independent of the implementation).
Of course we can assume clone is save, but if the implementation changes this test will catch errors here.

Reply:
Understood, I've added some tests, are they sufficient or should I add some other test cases?